pkg/kubectl/cmd/cp_test.go

```diff
@@ -497,3 +506,76 @@ func TestClean(t *testing.T) {
 		}
 	}
 }
+
+func TestCopyToPod(t *testing.T) {
+	tf := cmdtesting.NewTestFactory()
+	tf.Namespace = "test"
+	ns := legacyscheme.Codecs
+	codec := legacyscheme.Codecs.LegacyCodec(scheme.Versions...)
+
+	tf.Client = &fake.RESTClient{
+		GroupVersion:         schema.GroupVersion{Group: "", Version: "v1"},
+		NegotiatedSerializer: ns,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
```

Comment:
@soltysh 
Did you notice that I have modified the client server here. Now the server will response with http.StatusNotFound. So I think we are sure that `the server could not find the requested resource` equals that the request sent correctly.
From my understanding, what you mean is that we should just check if the failure is  FilePathEmpty error, if not, it indicates that the request has been sent correctly.
Anyway, I have updated my patch as my understanding.
If I misunderstand what you mean, please correct me. Thanks.